app/controllers/admin_incoming_message_controller.rb

```diff
@@ -36,6 +36,26 @@ def destroy
     redirect_to admin_request_url(@incoming_message.info_request)
   end
 
+  def bulk_destroy
+    if params[:commit] == "No"
+      redirect_to(admin_request_url(params[:request_id]))
+    end
+
+    @incoming_messages = IncomingMessage.where(:id => params[:ids].split(","))
+    if params[:commit] == "Yes"
+      info_request = InfoRequest.find(params[:request_id])
+      @incoming_messages.each do |message|
+        message.destroy
+        info_request.log_event("destroy_incoming",
+                               { :editor => admin_current_user,
+                                 :deleted_incoming_message_id => message.id })
+      end
+      info_request.expire
```

Comment:
Hmm (consider this a note to self) - wonder if deleting an individual message through the existing delete button does the same thing